src/PrintCore/PrintCore.cs

```diff
@@ -0,0 +1,775 @@
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMCreateSession (out IntPtr session);
+
+		internal PMPrintSession (IntPtr handle, bool owns) : base (handle, owns) {}
+		
+		public PMPrintSession ()
+		{
+			IntPtr value;
+			var code = PMCreateSession (out value);
+			if (code == PMStatusCode.Ok)
+				handle = value;
+			else
+				throw new PMPrintException (code);
+		}
+
+		public static PMStatusCode TryCreate (out PMPrintSession session)
+		{
+			IntPtr value;
+			var code = PMCreateSession (out value);
+			if (code == PMStatusCode.Ok){
+				session = new PMPrintSession (value, true);
+				return PMStatusCode.Ok;
+			}
+			session = null;
+			return code;
+		}
+
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionError (IntPtr handle);
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionSetError (IntPtr handle, PMStatusCode code);
+		
+		public PMStatusCode SessionError {
+			get {
+				return PMSessionError (handle);
+			}
+			set {
+				PMSessionSetError (handle, value);
+			}
+		}
+		
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionDefaultPrintSettings (IntPtr session, IntPtr settings);
+
+		public void AssignDefaultSettings (PMPrintSettings settings)
+		{
+			if (settings == null)
+				throw new ArgumentNullException (nameof (settings));
+			PMSessionDefaultPrintSettings (handle, settings.handle);
+		}
+
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionDefaultPageFormat (IntPtr session, IntPtr pageFormat);
+		
+		public void AssignDefaultPageFormat (PMPageFormat pageFormat)
+		{
+			if (pageFormat == null)
+				throw new ArgumentNullException (nameof (pageFormat));
+			PMSessionDefaultPageFormat (handle, pageFormat.Handle);
+		}
+		
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMSessionCreatePrinterList (IntPtr printSession, out IntPtr printerListArray, out int index, out IntPtr printer);
+
+		internal static string [] FetchArray (IntPtr cfArrayString, bool owns)
+		{
+			string [] ret;
+			// Fetc the values from the array that we own.
+			var arr = new CFArray (cfArrayString, owns: owns);
+			int c = (int) arr.Count;
+			ret = new string [c];
+			for (int i = 0; i < c; i++)
+				ret [i] = CFString.FetchString (arr.GetValue (i));
+			arr.Dispose ();
+
+			return ret;
+		}
+		
+		public PMStatusCode CreatePrinterList (out string [] printerList, out int index, out PMPrinter printer)
+		{
+			IntPtr array, printerHandle;
+			var code = PMSessionCreatePrinterList (handle, out array, out index, out printerHandle);
+			if (code != PMStatusCode.Ok){
+				printerList = null;
+				printer = null;
+				return code;
+			}
+
+			printerList = FetchArray (array, true);
+			if (printerHandle != IntPtr.Zero){
+				// Now get the printer, we do not own it, so retain.
+				NativeInvoke.PMRetain (printerHandle);
+				printer = new PMPrinter (printerHandle, owns: false);
```

Comment:
It would be simpler to pass `owns: true` instead of calling `PMRetain` just before and passing `owns: false`.